### ISSUE TITLE
SDL2 version of the example (or even better, backend-independent)

### ISSUE THREAD

**Issue by cesss** [Immediate-Mode-UI/Nuklear#495]
The best screenshots at the Readme come from the demo in the "example" folder. However, that demo is not just a "demo", because it includes a complete GLFW backend, thus making that example GLFW-dependent and GLEW-dependent, which is unfortunate. I believe that example could be rewritten in a backend-independent way, just like the small demos in the source tree.

The unfortunate thing is not only making it dependent on a specific backend, but, by implementing the backend together with the example, the code is unnecessarily complex. The small demos separate the backend and the demo code, making it very easy to understand (and not because they are smaller, but because you don't mix what belongs to the backend, and what belongs to the demo).

I prefer Nuklear over ImGui because I use C, but in ImGui you can see the complete demo in all backends. In Nuklear it should be possible to. Those of us who don't use GLFW, miss the fun of the best example...

**Comment by dumblob** [comment]
I totally agree. We just do not have enough resources to implement it. Feel free to make a PR, it surely will get merged.

Also if you wanted, I will grant you admin rights to increase the current [bus factor](https://en.wikipedia.org/wiki/Bus_factor ) of ~3 after you [reviewed](https://github.com/Immediate-Mode-UI/Nuklear#reviewers-guide ) a few of the current PRs of others.

**Comment by cesss** [comment]
Well, step by step. What I thought is not to port the complete example folder, because it would be a lot of work. Instead, I was thinking in porting component by component. For example, starting with the buttons dialog.

**Comment by dumblob** [comment]
Yep, that is fine. Anything is fine if it moves Nuklear forward :wink:.
